src/main/kotlin/day05/Wand.kt

```
@@ -0,0 +1,29 @@
+package org.example.day05
+
+class Wand(
+    _name: String?,
+    _power: Double,
+) {
+    var name: String? = if(_name == null || _name.length < 3) throw IllegalArgumentException() else _name // 이야 컴파일러가 똑똑해서 여기서 이미 null값 걸러지는 줄 알고서는 setter에서 타입을 바꿔버리네!
```

detekt (1.23.7)

[warning] A call to the default constructor of an exception was detected. Instead one of the constructor overloads should be called. This allows to provide more meaningful exceptions.

(detekt.exceptions.ThrowingExceptionsWithoutMessageOrCause)